File: assign2/ocaml/src/iterative.ml
let iterativeFib position =
  let fib_iter (prev, cur) _ = (cur, prev + cur) in

  let fibs = List.init (position + 1) (fun pos -> 
    if pos = 0 then 
      (1, 1)
    else 
      List.fold_left fib_iter (1, 1) (List.init pos (fun _ -> ()))
      ) in

  List.map fst fibs

File: assign2/erlang/src/iterative.erl
-module(iterative).
-export([iterativeFib/1]).

iterativeFib(0) -> [1];
iterativeFib(1) -> [1, 1];
iterativeFib(Position) when Position >= 2 ->
    {Series, _} = lists:foldl(
        fun(_, {Series, {Prev, Cur}}) ->
            {Series ++ [Cur], {Cur, Prev + Cur}}
        end,
        {[1, 1], {1, 2}},
        lists:seq(2, Position)
    ),

    Series.
